src/main/java/org/spongepowered/common/mixin/core/item/inventory/MixinContainerCrafting.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.core.item.inventory;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.ClickType;
+import net.minecraft.inventory.Container;
+import net.minecraft.inventory.ContainerPlayer;
+import net.minecraft.inventory.ContainerWorkbench;
+import net.minecraft.inventory.IContainerListener;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import org.spongepowered.asm.mixin.Mixin;
+
+@Mixin(value = {
+        ContainerWorkbench.class,
+        ContainerPlayer.class
+})
+public abstract class MixinContainerCrafting extends MixinContainer {
+
+    private final Container this$ = (Container) (Object) this;
+
+    @Override
+    public ItemStack slotClick(int slotId, int dragType, ClickType clickTypeIn, EntityPlayer player) {
+        ItemStack result = super.slotClick(slotId, dragType, clickTypeIn, player);
+
+        if (slotId == 0) {
+            // Clicked the result
+            for (IInventory inventory : inventory$getInventory().allInventories()) {
+                for (IContainerListener listener : this.listeners) {
+                    //listener.sendAllWindowProperties(this$, inventory);
+                    listener.updateCraftingInventory(this$, getInventory());
+                }
+            }
+        } else {
+            for (IContainerListener listener : this.listeners) {
+                listener.sendSlotContents(this$, 0, this.getInventory().get(0));
```

Comment:
Isn't triggered when the player clicks the item in the result slot, so I ended up doing this instead.